README.md

```diff
@@ -93,9 +93,64 @@ Currently, we support only Windows (7 or later), Linux and unofficially macOS.
 
 ## Build and Installation
 
+### Build in Docker
+
+#### Build image
+
+Building in Docker doesn't require installation of required libraries locally.
+This is a good option for trying out retdec without setting up the whole build toolchain.
+
+To build the retdec docker image, run
+
+```
+docker build -t retdec .
+```
+
+This builds the container from the master branch of this repository.
+
+To build the container using the local copy of the repository, fully clone the repository:
+
+```
+git submodule update --init --recursive
+```
+
+Then build the container using the development Dockerfile, `Dockerfile.dev`:
+
+```
+docker build -t retdec:dev . -f Dockerfile.dev
+```
+
+#### Run container
+
+To decompile a binary, create a container to upload the binary to:
+
+```
+docker create --name retdec retdec
+```
+
+Upload the binary:
+
+```
+docker cp <file> retdec:/destination/path/of/binary
+```
+
+Run the decompiler:
+
+```
+docker run retdec decompile.sh /destination/path/of/binary
```

Comment:
This process does not work:
```
$ docker create --name retdec retdec
464c16d576d6acda87caed2e1143460a958dcc57571be51b0328be7c39f3a6fd
$ docker cp test.exe retdec:/home/retdec/test.exe
$ docker run retdec decompile.sh /home/retdec/test.exe
Error: The input file '/home/retdec/test.exe' does not exist or is not readable
```
I believe that the problem is that `docker cp` copies the file into the `retdec` _image_ but `docker run` runs the command in the `retdec` _container_, which does not have the file. Only after I committed the image via `docker commit`, I was able to decompile the file in the container that the commit produced.

What should be the correct process of using the container?

Full disclosure: I am a Docker beginner, so feel free to correct me when I did something wrong. However, I followed the instructions that are provided in the README.

Reply:
I must have omitted the `docker commit` step in the readme. 
Fixed and tested. 
I think you have terminology backwards; `docker cp` copies into the created container, but `docker run` runs images. `docker commit` turns the container into an image. 
I am also a Docker beginner, using little projects like this to become more comfortable.